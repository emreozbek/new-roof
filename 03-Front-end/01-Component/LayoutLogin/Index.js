import React, { Component } from "react";
import PropTypes from "prop-types";
import { Grid, withStyles } from "material-ui";

import styles from "./Style";

class LayoutLogin extends Component {
  render() {
    const { classes, children } = this.props;
    return (
      <Grid
        container
        column="column"
        justify="center"
        alignItems="center"
        spacing={0}
      >
        <div className={classes.background} />
        <Grid item className={classes.grid}>
          {children}
        </Grid>
      </Grid>
    );
  }
}
LayoutLogin.propTypes = {
  classes: PropTypes.object.isRequired,
  children: PropTypes.element.isRequired
};

export default withStyles(styles)(LayoutLogin);
