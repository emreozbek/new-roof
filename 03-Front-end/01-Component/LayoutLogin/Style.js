const styles = theme => ({
  background: {
    display: "block",
    width: "100%",
    height: "50%",
    background: theme.palette.secondary["700"],
    position: "absolute",
    left: 0,
    top: 0,
    zIndex: 0
  },
  grid: {
    zIndex: 10,
    maxWidth: 420
  },
  logo: {
    color: theme.palette.shades.dark.text.primary
  }
});
export default styles;
