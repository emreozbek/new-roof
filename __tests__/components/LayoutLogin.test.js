import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import LayoutLogin from "../../Client/Component/Layouts/Login/Index";

configure({ adapter: new Adapter() });

it("renders correctly", () => {
  const layoutLogin = shallow(<LayoutLogin />);
  expect(layoutLogin).toMatchSnapshot();
});
