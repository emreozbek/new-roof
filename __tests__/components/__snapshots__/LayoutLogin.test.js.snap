// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<LayoutLogin
  classes={
    Object {
      "background": "LayoutLogin-background-1",
      "grid": "LayoutLogin-grid-2",
      "logo": "LayoutLogin-logo-3",
    }
  }
/>
`;
